### PR TITLE
Add shipstation connector

### PR DESCRIPTION
**Pull request by Chuhaa**
# Description
Add shipstation connector

Fixes # ([issue](https://github.com/ballerina-platform/ballerina-extended-library/issues/287))

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

# Checklist:

### Changes to OpenAPI definition

- [x] Modified `info` section of the OpenAPI definition file - [Example](https://github.com/ballerina-platform/ballerina-extended-library/discussions/74)

    - `description` - what the connector is about 
    - `x-ballerina-init-description` OpenAPI extension - connector initialization details 
    - `x-ballerina-display` OpenAPI extension - connector name and path to the connector icon 

- [x] Added proper description each API Key, if API Key authentication is defined in OpenAPI definition file 

### Changes to connector module

- [x] Added license header at the top of each .bal file. 
- [x] Added Package.md as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/77)
- [x] Added Module.md for the module of the  connector as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/78)
- [x] Added connector icon to the root of the connector. Icon needs to be a `png` of 200x200 px size with name `icon.png`
- [x] Added Ballerina.toml file with following information. For keywords refer to [guide](https://github.com/ballerina-platform/ballerina-extended-library/discussions/72)

### Verifying connector module

- [x] Compiled the connector successfully with the targeted ballerina version. 
- [x] Conducted smoke tests on the connector (Recommended when OpenAPI definition is obtain from an unofficial source)

### Security checks
 - [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: openapi/shipstation/Dependencies.toml [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+]
+
+[[package]]
+org = "ballerinax"
+name = "shipstation"
+version = "1.0.0"
+dependencies = [
+	{org = "ballerina", name = "http"},
+	{org = "ballerina", name = "log"},
+	{org = "ballerina", name = "test"},
```
Comment (aneeshafedo) [inline openapi/shipstation/Dependencies.toml:294]: It seems that the connector package does not directly use ballerina `log` and `test` modules. Shall we rebuild the project and push the clean Dependencies.toml ?

File: openapi/shipstation/client.bal [hunk 2]
```diff
@@ -0,0 +1,768 @@
+import ballerina/http;
```
Comment (aneeshafedo) [inline openapi/shipstation/client.bal:1]: Missing license header.

File: openapi/shipstation/client.bal [hunk 3]
```diff
@@ -0,0 +1,768 @@
...
+# ``429`` | Too Many Requests - Exceeded ShipStation API limits. When the limit is reached, your application should stop making requests until X-Rate-Limit-Reset seconds have elapsed.
+# ``500`` | Internal Server Error - ShipStation has encountered an error.
+# 
+# ## DateTime Format and Time Zone
+# 
+# ShipStation uses the ISO 8601 combined format for dateTime stamps being submitted to and returned from the API. Please be sure to submit all dateTime values as follows:
+# 
+# yyyy-mm-dd hh:mm:ss (24 hour notation). Example - ``2016-11-29 23:59:59``
+# 
+# The time zone represented in all API responses is PST/PDT. Similarly, ShipStation asks that you make all time zone convertions and submit any dateTime requests in PST/PDT.
```
Comment (aneeshafedo) [inline openapi/shipstation/client.bal:182]: Connector description is very lengthy. Will this appear nicely in Ballerina central ? @abeykoon any thoughts ?

File: openapi/shipstation/client.bal [hunk 4]
```diff
@@ -0,0 +1,768 @@
...
+    # + clientConfig - The configurations to be used when initializing the `connector` 
+    # + serviceUrl - URL of the target service 
+    # + return - An error if connector initialization failed 
+    public isolated function init(ClientConfig clientConfig, string serviceUrl = "https://ssapi.shipstation.com/") returns error? {
+        http:Client httpEp = check new (serviceUrl, clientConfig);
+        self.clientEp = httpEp;
+        return;
+    }
+    # Register Account
+    #
```
Comment (aneeshafedo) [inline openapi/shipstation/client.bal:199]: Is there a possibility to add proper doc comments for remote functions ?

File: openapi/shipstation/types.bal [hunk 5]
```diff
@@ -0,0 +1,1444 @@
+public type ListTagsresponseArr ListTagsresponse[];
```
Comment (aneeshafedo) [inline openapi/shipstation/types.bal:1]: License header is missing

File: openapi/shipstation/utils.bal [hunk 6]
```diff
@@ -0,0 +1,223 @@
+import ballerina/url;
```
Comment (aneeshafedo) [inline openapi/shipstation/utils.bal:1]: License header is missing

File: openapi/shipstation/Dependencies.toml (hunk 1 above)
Comment (Chuhaa) [reply openapi/shipstation/Dependencies.toml]: yes. Fixed

File: openapi/shipstation/client.bal (hunk 2 above)
Comment (Chuhaa) [reply openapi/shipstation/client.bal:17]: Added

File: openapi/shipstation/client.bal (hunk 3 above)
Comment (Chuhaa) [reply openapi/shipstation/client.bal]: Reduced the length.

File: openapi/shipstation/client.bal (hunk 4 above)
Comment (Chuhaa) [reply openapi/shipstation/client.bal:71]: Updated the comments

File: openapi/shipstation/utils.bal (hunk 6 above)
Comment (Chuhaa) [reply openapi/shipstation/utils.bal:17]: Added

**Review by aneeshafedo — APPROVED** [review]

**Pull request merged**